app/controllers/queues_controller.rb

```diff
@@ -52,8 +53,38 @@ def spam_complete
   end
 
   def suggested_edit
+    @suggested_edit = SuggestedEdit.find_by(reviewed_by_id: nil)
+    if @suggested_edit.nil?
+      flash[:info] = "There are no suggested edits to review!"
+      queue_complete_redirect
+    else
+      @request = @suggested_edit.resource
+    end
   end
 
   def suggested_edit_complete
+    @suggested_edit = SuggestedEdit.find(params[:suggested_edit_id])
+    if params[:reject]
+      @suggested_edit.result = "reject"
+      @suggested_edit.reviewed_by_id = current_user.id
+    elsif params[:approve]
+      @suggested_edit.result = "approve"
+      @suggested_edit.reviewed_by_id = current_user.id
+      rescue_request = @suggested_edit.resource
+      flash[:warning] = "Edits unable to be committed. " unless rescue_request.update(@suggested_edit.new_values)
+    end
+    flash[:danger] = "Unable to save suggested edit review." unless @suggested_edit.save
+    if SuggestedEdit.all.count > 0
+      redirect_to action: :suggested_edit
+    else
+      flash[:sucess] = "No more suggested edits to review!"
+      redirect_to disaster_request_path(disaster_id: @suggested_edit.resource.disaster_id)
+    end
+  end
+
+  private
+
+  def queue_complete_redirect
+    request.referer == request.original_url ? redirect_to(:disasters) : redirect_back(fallback_location: :disasters)
```

Comment:
There's still a case in here that throws it into a redirect loop. Just avoid using `redirect_back` where possible, it's only useful in a very limited number of cases. Redirect to a defined location instead.

Reply:
But I'd like to simply bring them to wherever they were before they entered the queue. At the least, I'd like them to go the request index.

Reply:
Also, how does that get thrown into a loop?

Reply:
Best to just redirect them to requests index.

As for how - if the referer is equal to the original url but the client doesn't send the referer header, then request.referer is nil. Conditional returns false, we hit redirect_back, which redirects to the same page and tries to redirect again.